src/vaadin-context-menu.html

```diff
@@ -317,6 +319,7 @@
 
             /**
              * Event name to listen for closing the context menu.
+             * @type {string}
              */
             closeOn: {
```

Comment:
Should this accept nullish values?
`_closeOnChanged` handles this explicitly by listening to `vaadin-overlay-outside-click`.

Reply:
I don't think so. This has a default string value and you're supposed to set a name of an event to it. Imo it's better that you can only set a string to it if using TS (API is more clear), but still good to have the checks in implementation so the code is safer even for non-TS users.

Reply:
Emtpy string is valid though I guess.
`_templateOrRendererChanged()` has one line where it sets `this.closeOn = '';`

So the check `if (closeOn) {` caches that.

Reply:
If clearing the property is a valid thing to do and `''` is accepted for this purpose, I don't see why nullish values should not.

Reply:
Actually now that I think about it (both `closeOn` and `openOn`) if I'd like to prevent it from closing/opening on some event maybe it would be nicer to be able to set it to `null` or `undefined` instead of just empty string `''`.

I'm not sure which would be better though `null` or `undefined`. Maybe it would be better to just leave this `@type` out for both `closeOn` and `openOn` so it will generate `string|null|undefined`.

Reply:
I think we can add `undefined` as allowed value. But I would prefer `string` here.

Ideally it should be something like `keyof HTMLElementEventMap`.

Reply:
I don't have strong opinions about it. I'm ok with either omitting `@type`, having `@type {string}` or `@type {string | undefined}`. @pekam can decide about marking this as resolved or not.

Reply:
I haven't changed my personal opinion. Clearing the prop is explicitly handled valid case, so I'd expect that I can use null and undefined for this purpose.

But it's not a big deal to me either, and I also trust @web-padawan's judgement on this. So I'll resolve the comment.